src/app/main/header/header.component.ts

```diff
@@ -45,6 +57,11 @@ export class HeaderComponent {
         });
     }
 
+    ngOnInit() {
+        this.componentCommsSubscription = this._componentCommsService.on(
```

Comment:
you should unsubscribe to the subscription

Reply:
good catch! fixed in 73b3db3